src/MonitorProfiler/ProbeInstrumentation/ProbeInjector.cpp

```diff
@@ -138,6 +157,70 @@ HRESULT ProbeInjector::InstallProbe(
     pNewInstr->m_Arg32 = request.pAssemblyData->GetProbeMemberRef();
     rewriter.InsertBefore(pInsertProbeBeforeThisInstr, pNewInstr);
 
+    pNewInstr = pNewInstr = rewriter.NewILInstr();
+    pNewInstr->m_opcode = CEE_LEAVE;
+    pNewInstr->m_pTarget = pInsertProbeBeforeThisInstr;
+    rewriter.InsertBefore(pInsertProbeBeforeThisInstr, pNewInstr);
+
+    // END: Try block
+    // START: Catch block
+
+    pCatchBegin = rewriter.NewILInstr();
+    pCatchBegin->m_opcode = CEE_POP;
+    rewriter.InsertBefore(pInsertProbeBeforeThisInstr, pCatchBegin);
+
+    // START: Try block (nested)
+
+    pNestedTryBegin = rewriter.NewILInstr();
+    pNestedTryBegin->m_opcode = CEE_LDC_I8;
+    pNestedTryBegin->m_Arg64 = request.uniquifier;
+    rewriter.InsertBefore(pInsertProbeBeforeThisInstr, pNestedTryBegin);
+
+    pNewInstr = rewriter.NewILInstr();
+    pNewInstr->m_opcode = CEE_LDC_NATIVE_I;
+    pNewInstr->m_Arg64 = reinterpret_cast<INT64>(pFaultingProbeCallback);
+    rewriter.InsertBefore(pInsertProbeBeforeThisInstr, pNewInstr);
+
+    pNewInstr = rewriter.NewILInstr();
+    pNewInstr->m_opcode = CEE_CALLI;
+    pNewInstr->m_Arg32 = request.pAssemblyData->GetFaultingProbeCallbackSignature();
+    rewriter.InsertBefore(pInsertProbeBeforeThisInstr, pNewInstr);
+
+    pNestedTryLeave = rewriter.NewILInstr();
+    pNestedTryLeave->m_opcode = CEE_LEAVE;
+    rewriter.InsertBefore(pInsertProbeBeforeThisInstr, pNestedTryLeave);
+
+    // END: Try block (nested)
+    // START: Catch block (nested)
```

Comment:
nit: Just for consistency with the comments, do you want to add in `END: Catch block (nested)` and `END: Catch block`